include/bh_python/register_axis.hpp

```diff
@@ -114,6 +114,41 @@ py::class_<A> register_axis(py::module& m, Args&&... args) {
                  }
              })
 
+        .def_property_readonly("traits_underflow",
+                               [](const A& self) {
+                                   return static_cast<bool>(
+                                       bh::axis::traits::options(self)
```

Comment:
`bh::axis::traits::options(self)` just calls `.options()` on a dynamic axes. You don't need to go through the extra layer of abstraction here, I think. Calling `self.options()` directly is more efficient during compilation.

Reply:
Some additional explanation: The `axis::traits` exist as an indirection layer so that Boost.Histogram can accept any class that supports the minimal required Axis interface. But here you know for sure that `self` is a `boost::histogram::axis::variant`, so you don't need the extra layer of indirection.

Reply:
Fantastic, thanks!